### PR TITLE
ZEN-31990: Add Impact upgrade to cse-upgrade.sh

### PR DESCRIPTION
**Pull request by Deer-WarLord**

### CODE REVIEW

File: cse/upgrade-cse.sh.in [hunk 1]
```diff
@@ -51,4 +51,8 @@ serviced script run /root/%SHORT_VERSION%.x/current_version.txt --service Zenoss
 echo "Starting resmgr upgrade."
 serviced script run /root/%SHORT_VERSION%.x/upgrade-cse.txt --service Zenoss.cse
 
+if [ $(serviced service status --show-fields Name  Zenoss.cse/Infrastructure/Impact | grep Impact) ]; then
+    echo "Starting Impact upgrade."
+    serviced script run /root/%SHORT_VERSION%.x/upgrade-impact.txt --service Zenoss.cse
+
```
Comment (jpeacock-zenoss) [inline cse/upgrade-cse.sh.in]: Looks like you're missing a closing `fi` statement.

File: cse/upgrade-impact.txt.in [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/solr
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/redis
+SVC_WAIT {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-model {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-events {{ZENOSS_TOP_SVC}}/Infrastructure/RabbitMQ {{ZENOSS_TOP_SVC}}/Zenoss/Events/zeneventserver {{ZENOSS_TOP_SVC}}/Infrastructure/solr {{ZENOSS_TOP_SVC}}/Infrastructure/redis started 600
+
+# Stop Impact
+SVC_STOP {{ZENOSS_TOP_SVC}}/Infrastructure/Impact
+SVC_WAIT {{ZENOSS_TOP_SVC}}/Infrastructure/Impact stopped 60
+
+# Install upgraded impact service definition
+SVC_RUN {{ZENOSS_TOP_SVC}}/Zenoss/User\ Interface/Zope {{ZENPACK_COMMAND}} install {{ZENPACK_VERSION}}
```
Comment (jpeacock-zenoss) [inline cse/upgrade-impact.txt.in]: This PR seems to be incomplete.  There's no makefile that's replacing the `{{ZENOSS_TOP_SVC}}`, `{{ZENPACK_COMMAND}}`, and `{{ZENPACK_VERSION}}` placeholder variables.

Comment (Deer-WarLord) [reply cse/upgrade-impact.txt.in]: yes, you're right. I'll add those variables after manual testing of CZ upgrade. 
Looks like upgrade script makes the Impact upgrade even without these changes, so I need to find out whether it does it correctly

File: cse/upgrade-impact.txt.in [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+SVC_USE {{IMPACT_IMAGE}}
+
+# Start all our prerequisite services
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-model
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-events
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/RabbitMQ
+SVC_START {{ZENOSS_TOP_SVC}}/Zenoss/Events/zeneventserver
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/solr
+SVC_START {{ZENOSS_TOP_SVC}}/Infrastructure/redis
+SVC_WAIT {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-model {{ZENOSS_TOP_SVC}}/Infrastructure/mariadb-events {{ZENOSS_TOP_SVC}}/Infrastructure/RabbitMQ {{ZENOSS_TOP_SVC}}/Zenoss/Events/zeneventserver {{ZENOSS_TOP_SVC}}/Infrastructure/solr {{ZENOSS_TOP_SVC}}/Infrastructure/redis started 600
```
Comment (jpeacock-zenoss) [inline cse/upgrade-impact.txt.in]: I think all these services are already running when this script runs.

Also, if `SVC_START` is required, then the `memcached` service is missing, which provides mariadb-model caching for Python code.

File: cse/upgrade-cse.sh.in [hunk 4]
```diff
@@ -51,4 +51,8 @@ serviced script run /root/%SHORT_VERSION%.x/current_version.txt --service Zenoss
 echo "Starting resmgr upgrade."
 serviced script run /root/%SHORT_VERSION%.x/upgrade-cse.txt --service Zenoss.cse
 
+if [ $(serviced service status --show-fields Name  Zenoss.cse/Infrastructure/Impact | grep Impact) ]; then
```
Comment (jpeacock-zenoss) [inline cse/upgrade-cse.sh.in]: The `if` statement itself is probably unnecessary.  This is for Cloud only which means that Impact is always installed.

File: cse/upgrade-cse.sh.in [hunk 5]
```diff
@@ -51,4 +51,8 @@ serviced script run /root/%SHORT_VERSION%.x/current_version.txt --service Zenoss
...
+    echo "Starting Impact upgrade."
+    serviced script run /root/%SHORT_VERSION%.x/upgrade-impact.txt --service Zenoss.cse
```
Comment (jpeacock-zenoss) [inline cse/upgrade-cse.sh.in]: With the removal of the `if` statement, I think the code in `upgrade-impact.txt.in` can be integrated into `upgrade-cse.txt.in`.  Also, unless there's some special code in prodbin, the `upgrade-cse.txt.in` file will have already upgraded the Impact zenpacks.

**Pull request closed without merging**